src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -396,8 +395,9 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       case Failure(e) =>
         logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
         currentSender ! CallActor.Ack(message)
-        self ! CheckForWorkflowComplete
+        self ! CallFailedNonRetryable(callKey, events, returnCode, e.getMessage)
       case Success(_) =>
+        executionStore += keyClone -> ExecutionStatus.NotStarted
```

Comment:
This is not safe, it's modifying the executionStore in an async callback and not in the WA's message processing thread.